<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QGridLayout" name="layout">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <item row="0" column="0" colspan="2">
       <widget class="MyWidget" name="myWidget" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>8</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QGroupBox" name="viewPortGroupBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>1</horstretch>
          <verstretch>4</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>ViewPort</string>
        </property>
        <widget class="QWidget" name="layoutWidget">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>40</y>
           <width>341</width>
           <height>76</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="viewPortGridLayout">
          <item row="0" column="0">
           <layout class="QFormLayout" name="viewPortXFormLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="viewPortXLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>1</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>X:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="viewPortXSpinBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>1</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimum">
               <number>-1000</number>
              </property>
              <property name="maximum">
               <number>1000</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="0" column="1">
           <layout class="QFormLayout" name="viewPortYFormLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="viewPortYLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>1</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Y:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="viewPortYSpinBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>1</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimum">
               <number>-1000</number>
              </property>
              <property name="maximum">
               <number>1000</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <layout class="QFormLayout" name="viewPortWidthFormLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="viewPortWidthLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>1</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Width:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="viewPortWidthSpinBox">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>1</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimum">
               <number>-1000</number>
              </property>
              <property name="maximum">
               <number>1000</number>
              </property>
              <property name="value">
               <number>500</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="1">
           <layout class="QFormLayout" name="viewPortHeightFormLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="viewPortHeightLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>1</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Height:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="viewPortHeightSpinBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>1</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimum">
               <number>-1000</number>
              </property>
              <property name="maximum">
               <number>1000</number>
              </property>
              <property name="value">
               <number>300</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QGroupBox" name="windowGroupBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>1</horstretch>
          <verstretch>4</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Window</string>
        </property>
        <widget class="QWidget" name="layoutWidget">
         <property name="geometry">
          <rect>
           <x>14</x>
           <y>44</y>
           <width>321</width>
           <height>76</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="widowGridLayout">
          <item row="0" column="0">
           <layout class="QFormLayout" name="windowXFormLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="windowXLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>1</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>X:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="windowXSpinBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>1</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimum">
               <number>-1000</number>
              </property>
              <property name="maximum">
               <number>1000</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="0" column="1">
           <layout class="QFormLayout" name="windowYFormLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="windowYLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>1</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Y:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="windowYSpinBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>1</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimum">
               <number>-1000</number>
              </property>
              <property name="maximum">
               <number>1000</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <layout class="QFormLayout" name="windowWidthFormLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="windowWidthLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>1</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Width:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="windowWidthSpinBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>1</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimum">
               <number>-1000</number>
              </property>
              <property name="maximum">
               <number>1000</number>
              </property>
              <property name="value">
               <number>500</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="1">
           <layout class="QFormLayout" name="windowHeightFormLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="windowHeightLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>1</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Height:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="windowHeightSpinBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>1</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimum">
               <number>-1000</number>
              </property>
              <property name="maximum">
               <number>1000</number>
              </property>
              <property name="singleStep">
               <number>1</number>
              </property>
              <property name="value">
               <number>300</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item row="2" column="0" colspan="2">
       <widget class="QGroupBox" name="drawRectGroupBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>1</horstretch>
          <verstretch>2</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>DrawRect</string>
        </property>
        <widget class="QWidget" name="">
         <property name="geometry">
          <rect>
           <x>13</x>
           <y>33</y>
           <width>761</width>
           <height>36</height>
          </rect>
         </property>
         <layout class="QHBoxLayout" name="drawRectHLayout">
          <item>
           <layout class="QFormLayout" name="drawRectXFormLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="drawRectXLabel">
              <property name="text">
               <string>X:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="drawRectXSpinBox">
              <property name="minimum">
               <number>-1000</number>
              </property>
              <property name="maximum">
               <number>1000</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QFormLayout" name="drawRectYFormLayou">
            <item row="0" column="0">
             <widget class="QLabel" name="drawRectYLabel">
              <property name="text">
               <string>Y</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="drawRectYSpinBox">
              <property name="minimum">
               <number>-1000</number>
              </property>
              <property name="maximum">
               <number>1000</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QFormLayout" name="drawRectWidthFormLayou">
            <item row="0" column="0">
             <widget class="QLabel" name="drawRectWidthLabel">
              <property name="text">
               <string>Width:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="drawRectWidthSpinBox">
              <property name="minimum">
               <number>-1000</number>
              </property>
              <property name="maximum">
               <number>1000</number>
              </property>
              <property name="value">
               <number>100</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QFormLayout" name="drawRectHeightFormLayou">
            <item row="0" column="0">
             <widget class="QLabel" name="drawRectHeightLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Height:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="drawRectHeightSpinBox">
              <property name="minimum">
               <number>-1000</number>
              </property>
              <property name="maximum">
               <number>1000</number>
              </property>
              <property name="value">
               <number>100</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MyWidget</class>
   <extends>QWidget</extends>
   <header>mywidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
